msteams-platform/tabs/how-to/using-teams-client-sdk.md

```diff
@@ -1,72 +1,469 @@
+```TypeScript
+import { app, Context } from "@microsoft/teams-js";
 
-### Authentication namespace
+app.getContext().then((context: Context) => {
+    /*...*/
+});
+```
 
-| Function  | Description          | Documentation|
-| -----     | -----     | -----    |
-|`microsoftTeams.authentication.authenticate(authenticateParameters?: AuthenticateParameters)`|Initiates an authentication request that opens a new window with the parameters provided by the caller. Optional input values are defined by the **AuthenticateParameters** object.|[function](/javascript/api/@microsoft/teams-js/microsoftteams.authentication?view=msteams-client-js-latest&preserve-view=true)<br/>[auth obj](/javascript/api/@microsoft/teams-js/microsoftteams.authentication.authenticateparameters?view=msteams-client-js-latest&preserve-view=true)|
-|`microsoftTeams.authentication.notifySuccess(result?: string, callbackUrl?: string)`|Notifies the frame that initiated the authentication request that the request was successful and closes the authentication window|[function](/javascript/api/@microsoft/teams-js/microsoftteams.authentication?view=msteams-client-js-latest&preserve-view=true)|
-|`microsoftTeams.authentication.notifyFailure(reason?: string, callbackUrl?: string)`|Notifies the frame that initiated the authentication request that the request failed and closes the authentication window.|[function](/javascript/api/@microsoft/teams-js/microsoftteams.authentication?view=msteams-client-js-latest&preserve-view=true)|
-|`microsoftTeams.authentication.getAuthToken(authTokenRequest: AuthTokenRequest)`|Send request to issue Azure AD token on behalf of the app. The token can be acquired from the cache, if it has not expired. Otherwise, a request is sent to Azure AD to obtain a new token.|[function](/javascript/api/@microsoft/teams-js/microsoftteams.authentication?view=msteams-client-js-latest#@microsoft-teams-js-microsoftteams-authentication-getauthtoken&preserve-view=true)|
+...or the equivalent `async/await` pattern:
+
+```TypeScript
+import { app, Context } from "@microsoft/teams-js";
+
+async function example() {
+  const context: Context = await app.getContext();
+  /*...*/
+}
+```
+
+---
+
+> [!TIP]
+> When you use [Teams Toolkit to update to TeamsJS v.2.0](#updating-to-the-teams-client-sdk-v200), the required updates are flagged for you with `TODO` comments in your client code.
+
+### APIs organized into capabilities
+
+A *capability* is a logical grouping (via namespace) of APIs that provide similar functionality. You can think of Microsoft Teams, Outlook, and Office, as hosts to your tab app. A host supports a given capability if it supports all the APIs defined within that capability. A host can't partially implement a capability.  Capabilities can be feature- or content-based, such as *authentication*, or *dialog*. There are also capabilities for application types, such as *tabs/pages* or *bots*, and other groupings.
```

Comment:
nit: double space after the period -- "implement a capability." and "Capabilities can be feature-"